src/handlers/records-read.ts

```diff
@@ -68,6 +68,22 @@ export class RecordsReadHandler implements MethodHandler {
     if (matchedMessage.descriptor.method === DwnMethodName.Delete) {
       const recordsDeleteMessage = matchedMessage as RecordsDeleteMessage;
       const initialWrite = await RecordsWrite.fetchInitialRecordsWriteMessage(this.messageStore, tenant, recordsDeleteMessage.descriptor.recordId);
+
+      if (initialWrite === undefined) {
+        return messageReplyFromError(new DwnError(
+          DwnErrorCode.RecordsReadInitialWriteNotFound,
+          'Initial write for deleted record not found'
+        ), 400);
+      }
+
+      // Perform authorization before returning the delete and initial write messages
+      const parsedInitialWrite = await RecordsWrite.parse(initialWrite);
+      try {
+        await RecordsReadHandler.authorizeRecordsRead(tenant, recordsRead, parsedInitialWrite, this.messageStore);
+      } catch (error) {
+        return messageReplyFromError(error, 401);
+      }
```

Comment:
If this approach makes sense, I'll add tests next.

Reply:
@andresuribe87  🔥 Yup this is exactly what we were looking for! 🐐 🙏

Reply:
Added tests @LiranCohen 

Please take another look!

Reply:
Hey @andresuribe87, there is a question on how we handle authorship, will sync up with @LiranCohen again and get back to you!

Reply:
I think this is good to approve after talking with @LiranCohen, I'd like to rename some function arguments because the semantic of the write being passed for verification is no longer just the "matched RecordsWrite", but we can do that in a opportunistic clean up..